include/boost/url/impl/url_base.ipp

```diff
@@ -275,6 +324,57 @@ remove_authority() noexcept
         impl_.split(id_host, 0);
         impl_.split(id_port, 0);
     }
+    else if(encode_colon)
+    {
+        // encode colons in first segment
+        pos_t pn = impl_.len(id_path);
+        std::size_t cn = 0;
+        if (encode_colon)
+            for (char c: first_segment())
+                cn += (c == ':');
+        resize_impl(
+            id_user,
+            id_query,
+            pn + (2 * cn),
+            op);
+        impl_.split(id_user, 0);
+        impl_.split(id_pass, 0);
+        impl_.split(id_host, 0);
+        impl_.split(id_port, 0);
+        if (encode_colon)
+        {
+            // move the 2nd, 3rd, ... segments
+            auto begin = s_ + impl_.offset(id_path);
+            auto it = begin;
+            auto end = begin + pn;
+            while (*it != '/' &&
+                   it != end)
+                ++it;
+            std::memmove(it + (2 * cn), it, end - it);
```

Comment:
this needs to be `op.move` (or w/e it is called)

Reply:
https://github.com/boostorg/url/pull/581#discussion_r989332548